src/multi-select-typeahead/index.tsx

```diff
@@ -0,0 +1,263 @@
+	theme,
+	data: createDataMiddleware<ListOption>(),
+	focus
+})
+	.properties<MultiSelectTypeaheadProperties>()
+	.children<MultiSelectTypeaheadChildren>();
+
+export function arraysDifferent(arr1: string[], arr2: string[]): boolean {
+	if (arr1.length !== arr2.length) {
+		return true;
+	}
+
+	for (let i = 0; i < arr1.length; i++) {
+		if (arr1[i] !== arr2[i]) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+export const MultiSelectTypeahead = factory(function MultiSelectTypeahead({
+	middleware: { icache, theme, focus, data },
+	properties,
+	children
+}) {
+	const {
+		initialValue = [],
+		resource,
+		transform,
+		disabled,
+		itemsInView,
+		position,
+		name,
+		placement = 'inline'
+	} = properties();
+	const [{ label, items, selected } = {} as MultiSelectTypeaheadChildren] = children();
+	const themeCss = theme.classes(css);
+	const { value } = properties();
+	const { get, getOptions } = data();
+	const focused = icache.getOrSet('focused', false);
+
+	if (value !== undefined && arraysDifferent(value || [], icache.get('value') || [])) {
+		icache.set('value', value || []);
+	}
+
+	if (value === undefined && arraysDifferent(icache.getOrSet('initialValue', []), initialValue)) {
+		icache.set('value', initialValue);
+		icache.set('initialValue', initialValue);
+	}
+
+	const currentOptions = get(getOptions());
+	const chips = icache.getOrSet('value', []).map((value) => {
+		let option = find(currentOptions || [], (option) => option.value === value);
+
+		return (
+			<Chip
+				theme={theme.compose(
+					css,
+					chipCss
+				)}
+				key={`value-${value}`}
+				classes={{
+					'@dojo/widgets/chip': {
+						root: [themeCss.value]
+					}
+				}}
+				onClose={
+					disabled
+						? undefined
+						: () => {
+								const { onValue } = properties();
+								const values = [...icache.getOrSet('value', [])];
+
+								const valueIndex = values.indexOf(value);
+
+								if (valueIndex >= 0) {
+									values.splice(valueIndex, 1);
+									icache.set('value', values);
+
+									onValue && onValue(values);
+								}
+
+								focus.focus();
+						  }
+				}
+			>
+				{{
+					label: selected
+						? selected(value, option && option.label)
+						: (option && option.label) || value
+				}}
+			</Chip>
+		);
+	});
+
+	const values = icache.getOrSet('value', []);
+	const active = focused || values.length > 0;
+
+	return (
+		<div
+			key="root"
+			classes={[
+				theme.variant(),
+				themeCss.root,
+				values.length > 0 ? themeCss.hasValue : null,
+				focused ? themeCss.focused : null,
+				label ? themeCss.hasLabel : null
+			]}
+		>
+			{label && (
+				<Label
+					focused={focused}
+					active={active}
+					theme={theme.compose(
+						css,
+						labelCss
+					)}
+					classes={{
+						'@dojo/widgets/label': {
+							root: [themeCss.label]
+						}
+					}}
+				>
+					{label}
+				</Label>
+			)}
+			<Typeahead
+				key="typeahead"
+				theme={theme.compose(
+					css,
```

Comment:
should this have a prefix

Reply:
Is this the correct order? I thought it was `baseCss, overrideCss`

Reply:
Nope! Good catch. Flipped all the theme.composes